### ISSUE TITLE
Structure of The Cannon

### ISSUE THREAD

**Issue by annayqho** [annayqho/AnnaCannon#9]
# Main Code:

## Input:
- a rectangular block that is N x L of pixel values of continuum-normalized spectral intensities (training set)
- a parallel block of inverse variance values
- a block of N x K labels
- a rectangular block that is M x L of pixel values of continuum-normalized spectral intensities (test set)
- a parallel block of inverse variance values
- a value for the "polynomial order" of the model

## Output:
- a block of M x K labels
- a block of Q x L parameters (coefficients and scatters and so on)

# Continuum Wavelengths Code:

## Input:
- the first three inputs to the Main Code

## Output:
- a length-L mask of continuum True/False values

# Continuum Normalization Code:

## Input:
- a N x L or M x L block of pixel values
- a parallel block of inverse variance values
- the length-L continuum mask (from above)

## Output
- a N x L or M x L block of continuum-normalized intensities
- a parallel block of rescaled inverse variance values

# APOGEE Munging Code:

## Input:
- a URL, or maybe nothing

## Output:
- a length-L rest-frame wavelength vector
- a block N x L of training-set pixel intensities
- a parallel block of inverse variances
- a block N x K of labels
- a block M x L of test-set pixel intensities
- a parallel block of inverse variances

## Comments:
This code should get the data, shift to the rest frame, read the bad-pixel masks, and build the inverse variance vectors and package them all into rectangular blocks.

# Workflow:
- Run APOGEE Munging Code
- Run Continuum Wavelengths Code
- Run Continuum Normalization Code
- Run Main Code
